migrations/entitlements/migration_test.go

```diff
@@ -1867,244 +1867,3 @@ func TestMigrateDictOfValues(t *testing.T) {
 		ref.Authorization,
 	)
 }
-
-func TestMigrateDictOfWithTypeValueKey(t *testing.T) {
```

Comment:
This test is no longer valid/working, since the migration uses the "legacy" hashing functions to remove the reference-typed key, whereas this test stores the "new format" of the reference type; thus the removal fails.
I've added a proper test at the FVM, which stores a reference type in the "old format", and then uses the legacy-hashing function for key-removal.

Reply:
Is there any way to also keep a variant of this test here on the Cadence side?

Reply:
Obviously I don't want to cause extra work here,  but could we maybe just add a legacy value definition in addition to the existing one, and stub everything out, but only leave the encoding part, similar to how we do this for link values in the cap cons migration?

Reply:
actually, turned out to be easier: only needed to 'override' the `Encode`method of the existing `LegacyReferenceType`: https://github.com/onflow/cadence/pull/3013/commits/efd96205743b1df5bf4ed6cf6bff56162f03d736

Reply:
Awesome!